src/java.base/share/classes/java/io/ObjectInputStream.java

```diff
@@ -116,18 +116,18 @@
  * the object's most specific class.
  *
  * <p>For example to read from a stream as written by the example in
- * ObjectOutputStream:
+ * {@link ObjectOutputStream}:
  * <br>
- * <pre>
- *      FileInputStream fis = new FileInputStream("t.tmp");
- *      ObjectInputStream ois = new ObjectInputStream(fis);
- *
- *      int i = ois.readInt();
- *      String today = (String) ois.readObject();
- *      Date date = (Date) ois.readObject();
- *
- *      ois.close();
- * </pre>
+ * {@snippet lang="java" :
+ *     try (FileInputStream fis = new FileInputStream("t.tmp");
+ *          ObjectInputStream ois = new ObjectInputStream(fis)) {
+ *         String label = (String) ois.readObject();
+ *         LocalDateTime date = (LocalDateTime) ois.readObject();
+ *         System.out.println(label + ": " + date);
```

Comment:
would it be worth using `System.out.printf` here?

Reply:
Maybe printing is a distraction; there many preferences for output and the original did not include it.
Simple string concatenation is the most efficient, but the least parameterizable.